polymetis/tests/scripts/3_timestamps.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+import time
+from typing import Dict
+
+import torch
+
+from polymetis import RobotInterface
+import torchcontrol as toco
+
+WARMUP_STEPS = 5
+TOTAL_STEPS = 30
+
+
+class TimestampCheckController(toco.PolicyModule):
+    i: int
+    warmup_steps: int
+    total_steps: int
+
+    def __init__(self, hz, warmup_steps, total_steps):
+        super().__init__()
+
+        self.dt = torch.tensor(1.0 / hz)
+        self.warmup_steps = warmup_steps
+        self.total_steps = total_steps
+
+        self.i = 0
+        self.ts_prev = torch.zeros(2).to(torch.int32)
+
+    def forward(self, state_dict: Dict[str, torch.Tensor]):
+        ts = state_dict["timestamp"]
+
+        # Check timestamp
+        if self.i > self.warmup_steps:
+            t_diff = toco.utils.timestamp_diff_seconds(ts, self.ts_prev)
+            assert torch.allclose(t_diff, self.dt, atol=1e-3)  # millisecond accuracy
```

Comment:
Is this test dependent on the machine it's run on? E.g. do we expect this test to become flaky if CircleCI's VMs get slow for whatever reason?

Reply:
Yes which is why I'm setting a very large tolerance.
An alternative is that we can also test if t_diff > 0, but the current test would also ensure that the frequency provided in the metadata from the simulation client is correct so I went for it.
I'm fine with either, thoughts?

Reply:
sgtm